app/views/shared/_cancel.html.erb

```diff
@@ -2,6 +2,10 @@
   <% if user_signing_up? %>
     <%= link_to cancel_link_text, sign_up_cancel_path, method: :get, class: 'usa-button usa-button--unstyled' %>
   <% else %>
-    <%= link_to cancel_link_text, link || return_to_sp_cancel_path %>
+    <% if defined?(link_method) %>
```

Comment:
Wonder if this might have been a good candidate for the `button_or_link_to` helper.

https://github.com/18F/identity-idp/blob/0579d710766d1c488234f7a1601d5c74ddd42615/app/helpers/link_helper.rb#L17

The `usa-button--unstyled` logic isn't handled there, but (a) maybe it should be or (b) maybe it's fine to apply `usa-button--unstyled` to a link.

Reply:
The helper did come up while I was working on it, but it initially seemed like it would be more effort to make the helper work for this case.

Somewhat related, but I'd also like to re-work some of this template too